src/lib.rs

```diff
@@ -585,6 +589,42 @@ where
         Ok(file)
     }
 
+    /// Delete a closed file with the given full path, if exists.
+    pub fn delete_file_in_dir(
+        &mut self,
+        volume: &Volume,
+        dir: &Directory,
+        name: &str,
+    ) -> Result<(), Error<D::Error>> {
+        debug!(
+            "delete_file(volume={:?}, dir={:?}, filename={:?}",
+            volume, dir, name
+        );
+        let dir_entry = match &volume.volume_type {
+            VolumeType::Fat(fat) => fat.find_directory_entry(self, dir, name),
+        };
+
+        let dir_entry = match dir_entry {
```

Comment:
Can we shorten this to `let dir_entry = dir_entry.map_err(Error::FileNotFound)?`;

Reply:
I shorten it even further since I can directly evaluate it from the previous match